geoips/interfaces/base.py

```diff
@@ -0,0 +1,360 @@
+                "will raise a TypeError.",
+                DeprecationWarning,
+                stacklevel=1,
+            )
+            return plugin_module_to_obj(self, module, module_call_func=func.__name__)
+
+    def get_plugins(self):
+        """Get a list of plugins for this interface."""
+        # This is more complicated than needed. To simplify, we will need to remove the
+        # function name from entry points.
+        plugins = []
+        for ep in get_all_entry_points(self.entry_point_group):
+            module = import_module(ep.__module__)
+            if hasattr(module, "call"):
+                plugins.append(plugin_module_to_obj(module))
+            else:
+                plugins.append(
+                    plugin_module_to_obj(module, module_call_func=ep.__name__)
+                )
+
+        # plugins = [
+        #     self.create_plugin(import_module(ep.__module__),
+        #     module_call_func=ep.__name__)
+        #     for ep in get_all_entry_points(self.entry_point_group)
+        # ]
+        return plugins
+
+    def is_valid(self, name):
+        """Check that an interface is valid.
+
+        Check that the requested interface function has the correct call signature.
+        Return values should be as specified below, but are not programmatically
+        verified.
+
+        Parameters
+        ----------
+        name : str
+          Name of the interface to be validated
+
+        Returns
+        -------
+        bool
+          True if valid, False if invalid
+        '''
+        """
+        plugin = self.get_plugin(name)
+
+    def test_interface_plugins(self):
+        """Test the current interface by validating every Plugin.
+
+        Test this interface by opening every Plugin available to the interface. Then validate each plugin by calling
+        `is_valid` for each.
+
+        Returns:
+            A dictionary containing three keys: 'by_family', 'validity_check', 'func', and 'family'. The value for each
+            of these keys is a dictionary whose keys are the names of the Plugins.
+
+            - 'by_family' contains a dictionary of plugin names sorted by family.
+            - 'validity_check' contains a dict whose keys are plugin names and whose values are bools where `True`
+              indicates that the Plugin's function is valid according to `is_valid`.
+            - 'func' contains a dict whose keys are plugin names and whose values are the function for each Plugin.
+            - 'family' contains a dict whose keys are plugin names and whose vlaues are the contents of the 'family'
+              attribute for each Plugin.
+        """
+        plugin_names = self.get_plugins(sort_by="family")
+        output = {
+            "by_family": plugin_names,
+            "validity_check": {},
+            "family": {},
+            "func": {},
+        }
+        for curr_family in plugin_names:
+            for curr_name in plugin_names[curr_family]:
```

Comment:
This is also not yet fully implemented, correct?  Per #91 ?

Reply:
Yes, same as above. I'm intending to replace this with `all_valid()`, though, which would simply iterate through all available plugins and call `is_valid()` for each of them. Does that sound right to you?

Reply:
Yes